transcoder/basisu_transcoder.cpp

```diff
@@ -12138,6 +12452,463 @@ namespace basist
+				lo |= (unpack_uastc_bitShift( _base, _bitnum, _index, _anchor ) << 4U);	//	3bits
+				hi |= (unpack_uastc_bitMask( _bitnum, _index, _anchor ) << 4U);			//	3bits
+		return	(((uint16_t)hi) << 8U) | ((uint16_t)lo);
+	}
+
+	static const anchorArrayCasesName	ms_anchorArrayDict[ 1 + TOTAL_ASTC_BC7_COMMON_PARTITIONS3 + TOTAL_BC7_3_ASTC2_COMMON_PARTITIONS + TOTAL_ASTC_BC7_COMMON_PARTITIONS2 ];
+
+	static const partition	ms_PartitionDict[ 1 + TOTAL_ASTC_BC7_COMMON_PARTITIONS3 + TOTAL_BC7_3_ASTC2_COMMON_PARTITIONS + TOTAL_ASTC_BC7_COMMON_PARTITIONS2 ];
+
+	static const param ms_param_group[param_case::kLast];
+
+	static	unpack_uastc_weights	ms_last;	//	to monitor unhandled cases
+
+	static	void	compute( const param * _param, const uint8_t * _src, const partition * _pattern, uint8_t * _dst, uint8_t _weight_mask8, uint8_t _invert_subset_mask8 )
+	{
+		#if ( defined(__arm__) || defined(__aarch64__) )
+		const uint8_t *	param = (const uint8_t *)_param;
+		uint8x16_t	bitLoad = vld1q_u8( _src );											//	load 8x16
+		uint8x8x2_t	bitSrc{ vget_low_u8( bitLoad ), vget_high_u8( bitLoad ) };
+
+		uint8x16_t	bitByte0 = vld1q_u8( param + 0 );
+		uint8x16_t	bitByte1 = vld1q_u8( param + 16 );
+		uint16x8_t	bitCtrl0 = vreinterpretq_u16_u8( vshrq_n_u8( bitByte0, 4 ) );		//	remove bytePos
+		uint16x8_t	bitCtrl1 = vreinterpretq_u16_u8( vshrq_n_u8( bitByte1, 4 ) );		//	remove bytePos
+					bitByte0 = vandq_u8( bitByte0, vdupq_n_u8(0x0F) );					//	filter bytePos
+					bitByte1 = vandq_u8( bitByte1, vdupq_n_u8(0x0F) );					//	filter bytePos
+		uint8x8_t	word03 = vtbl2_u8( bitSrc, vget_low_u8( bitByte0 ) );				//	translate 8x8
+		uint8x8_t	word47 = vtbl2_u8( bitSrc, vget_high_u8( bitByte0 ) );				//	translate 8x8
+		uint16x8_t	word07 = vreinterpretq_u16_u8( vcombine_u8( word03, word47 ) );		//	get word 16x8
+		uint8x8_t	word8B = vtbl2_u8( bitSrc, vget_low_u8( bitByte1 ) );				//	translate 8x8
+		uint8x8_t	wordCF = vtbl2_u8( bitSrc, vget_high_u8( bitByte1 ) );				//	translate 8x8
+		uint16x8_t	word8F = vreinterpretq_u16_u8( vcombine_u8( word8B, wordCF ) );		//	get word 16x8
+		uint16x8_t	bitShift0 = vandq_u16( bitCtrl0, vdupq_n_u16( 0x000F ) );			//	filter Shift
+		uint16x8_t	bitShift1 = vandq_u16( bitCtrl1, vdupq_n_u16( 0x000F ) );			//	filter Shift
+		int16x8_t	shr0 = vnegq_s16( vreinterpretq_s16_u16( bitShift0 ) );				//	neg shift 16x8
+		int16x8_t	shr1 = vnegq_s16( vreinterpretq_s16_u16( bitShift1 ) );				//	neg shift 16x8
+					word07 = vshlq_u16( word07, shr0 );									//	align 16x8
+					word8F = vshlq_u16( word8F, shr1 );									//	align 16x8
+		uint16x8_t	bitNum0 = vshrq_n_u16( bitCtrl0, 8 );								//	filter bitNum
+		uint16x8_t	bitNum1 = vshrq_n_u16( bitCtrl1, 8 );								//	filter bitNum
+		uint16x8_t	bitMask0 = vshlq_u16( vdupq_n_u16(1), bitNum0 );					//	bitMask = (1U << n) - 1U
+		uint16x8_t	bitMask1 = vshlq_u16( vdupq_n_u16(1), bitNum1 );					//	bitMask = (1U << n) - 1U
```

Comment:
This fails on GCC without `-flax-vector-conversions`:

```
../subprojects/basis_universal/transcoder/basisu_transcoder.cpp: In function 'void basist::pack_astc_block_weights(uint8_t*, const uint8_t*, int, int)':
../subprojects/basis_universal/transcoder/basisu_transcoder.cpp:12010:80: note: use '-flax-vector-conversions' to permit conversions between vectors with differing element types or numbers of subparts
12010 |                                 uint8x8_t       rev8 = vqmovn_u16( vcombine_u16( rev8lohi, vdup_n_u8(0) ) );                    //      8bits in 4 u8 (clear lower 32)
      |                                                                    ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/basis_universal/transcoder/basisu_transcoder.cpp:12010:101: error: cannot convert 'uint8x8_t' to 'uint16x4_t'
12010 |                                 uint8x8_t       rev8 = vqmovn_u16( vcombine_u16( rev8lohi, vdup_n_u8(0) ) );                    //      8bits in 4 u8 (clear lower 32)
      |                                                                                            ~~~~~~~~~^~~
      |                                                                                                     |
      |                                                                                                     uint8x8_t
```

An explicit cast fixes it:

```suggestion
		uint16x8_t	bitMask0 = vshlq_u16( vdupq_n_u16(1), (int16x8_t)bitNum0 );				//	bitMask = (1U << n) - 1U
		uint16x8_t	bitMask1 = vshlq_u16( vdupq_n_u16(1), (int16x8_t)bitNum1 );				//	bitMask = (1U << n) - 1U
```